### PR TITLE
KAFKA-2455: Test Failure: kafka.consumer.MetricsTest > testMetricsLeak

### PR DESCRIPTION
**Pull request by jinxing64**
DelayedFetchMetrics are loaded dynamically in function "onExpiration";
use assertNull(DelayedFetchMetrics) to initialize DelayedFetchMetrics explicitly;

### CODE REVIEW

**Comment by ijuma** [comment]
@ZoneMayor Why not add `assertNotNull(DelayedFetchMetrics.consumerExpiredRequestMeter)` before `val countOfStaticMetrics = Metrics.defaultRegistry().allMetrics().keySet().size` in the test?

**Comment by ijuma** [comment]
That will force the initialisation of the metrics before we compute the static count.

**Comment by ijuma** [comment]
Nice catch by the way. We really appreciate the work you're doing on making our tests more table. :)

**Comment by jinxing64** [comment]
@ijuma 
using assertNull to initialize DelayedFetchMetrics now

**Comment by ijuma** [comment]
Thanks @ZoneMayor, does this new version also fix the issue for you? If so, LGTM.

**Comment by ijuma** [comment]
Can you please update the PR description to explain the final change? It becomes part of the commit that is merged, so it is good if it is up to date.

**Comment by jinxing64** [comment]
@ijuma 
Yes, this is the new version fix issue for me;
I will keep updating the PR description to explain final change : )

**Comment by jinxing64** [comment]
@ijuma  
Could it be merged now?

**Comment by ijuma** [comment]
LGTM cc @guozhangwang

File: core/src/test/scala/unit/kafka/metrics/MetricsTest.scala [hunk 1]
```diff
@@ -58,6 +58,8 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     // force creation not client's specific metrics.
     createAndShutdownStep("group0", "consumer0", "producer0")
 
+    assertNotNull(DelayedFetchMetrics)
```
Comment (guozhangwang) [inline core/src/test/scala/unit/kafka/metrics/MetricsTest.scala:62]: Add comment here: "this assertion is only used for creating the metrics for DelayedFetchMetrics, it should never fail, but should not be removed".

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk and 0.9.0. Thanks @ZoneMayor for fixing these test failures! They are very important to our Jenkins reliability.

**Pull request closed without merging**